architecture/core.rst

```diff
@@ -390,6 +390,186 @@ normal world supports scheduling of processes. Even on UP systems, supporting
 several trusted threads in optee_os helps normal world scheduler to be
 efficient.
 
+Core handlers for native interrupts
+===================================
+
+OP-TEE core offers means to implement native interrupt
+(see :ref:`native_foreign_irqs`) controller drivers and means for device
+drivers to configure and register handler functions for these interrupts.
+Interrupt handlers can be nested for when an interrupt controller
+exposes interrupts which signaling is multiplexed on an interrupt
+controlled by a parent interrupt controller.
+
+Interrupt controllers are represented by an instance of ``struct itr_chip``.
+An interrupt controller exposes a given number of interrupts, each identified
+by an index from 0 to N-1 where N is the total number of interrupts exposed
+by that controller. In the litterature, an interrupt index identifier
```

Comment:
s/litterature/literature/

Reply:
@jforissier and I should take my English lessons :)

Reply:
To be fair, I'm just as much native English speaker as you are 😄

Reply:
But you're still a good reference for me. :beers: